netprobify/protocol/udp_unreachable.py

```diff
@@ -136,10 +137,21 @@ def generate_packets(self, all_groups, id_gen, logging_level="WARNING"):
                 log_udp_unreachable.debug(
                     "no source address found in group %s to reach %s", grp.name, self.destination
                 )
+
+            src_subnet = (
+                grp.src_subnet_ipv4 if self.address_family == "ipv4" else grp.src_subnet_ipv6
+            )
+            src_network = ip_network(src_subnet) if src_subnet else None
```

Comment:
```suggestion
            src_network = ip_network(get_src_subnet(self, grp))
```
and add `get_src_subnet` function in your UDP class (or Target maybe as it could be reused for other probing type) doing the following : 
```
def get_src_subnet(self, grp):
  if self.address_family == "ipv4":
    src_subnet = grp.src_subnet_ipv4 if self.address_family == "ipv4" else grp.src_subnet_ipv6
  else:
    src_subnet = grp.src_subnet_ipv6
  return src_subnet if src_subnet else None
```

Reply:
you are right, but why not return directly ip_network object directly?

Reply:
yes it would be even better

Reply:
done